### PR TITLE
Use context manager for xarray dataset file opening

### LINKED ISSUE #920: Loading TCTracks with `from_ibtracs_netcdf` does not properly close the filestream
**Describe the bug**
When reading the `IBTrACS.ALL.v04r00.nc` file with `from_ibtracs_netcdf()` the file is opened with `xr.open_dataset(ibtracs_path)` which opens a stream to the file that is not closed afterwards. I had problems with this on euler when trying to access the file at later stages with the following error:

`OSError: [Errno -101] NetCDF: HDF error: '/cluster/work/climate/sjuhel/climada/data/IBTrACS.ALL.v04r00.nc'`

**To Reproduce**
I haven't been able to reproduce the issue on a local computer. This is possibly related to NetCDF too.

**Expected behavior**
The file is properly closed once the data is loaded.

See also: https://github.com/pydata/xarray/issues/2887 

**Climada Version:** 4.1.1

**System Information (please complete the following information):**
 - Operating system and version: Ubuntu 22.04 (euler)
 - Python version: 3.10.13 (Custom venv while climada is not yet installed on the cluster, possibly part of the problems)

**Additional context**
I don't think this is a critical problem, but the following would probably be a better way to handle the opening of the file:

From:

https://github.com/CLIMADA-project/climada_python/blob/8f89ce10c3eaa14811c8c35247cda58de9c5516b/climada/hazard/tc_tracks.py#L469

to:

```python
with xr.open_dataset(ibtracs_path) as ds:
    ibtracks_ds = ds.load()
```

### PR DESCRIPTION
**Pull request by spjuhel**
Changes proposed in this PR:
Replace all occurrence of `data = xr.open_dataset(filename)` with `with xr.open_dataset(filename) as data:` to ensure filestream closing. See #920 on why this can cause problems.

Using `data = xr.open_dataset(filename)` assumes a data.close() afterwards to correctly close the filestream. This is not done in all function. The pythonic approach is to use a context manager `with ... as ... :` which handle the closing.

See also: https://github.com/pydata/xarray/issues/2887

This PR fixes #920 

### PR Author Checklist

- [x] Read the [Contribution Guide][contrib]
- [x] Correct target branch selected (if unsure, select `develop`)
- [x] Descriptive pull request title added
- [x] Source branch up-to-date with target branch
- [x] [Documentation](https://climada-python.readthedocs.io/en/latest/guide/Guide_PythonDos-n-Donts.html#2.--Commenting-&-Documenting) updated
- [x] [Tests][testing] updated
- [x] [Tests][testing] passing
- [x] No new [linter issues][linter]
- [x] [Changelog](https://github.com/CLIMADA-project/climada_python/blob/develop/CHANGELOG.md) updated

### PR Reviewer Checklist

- [ ] Read the [Contribution Guide][contrib]
- [ ] [CLIMADA Reviewer Checklist](https://climada-python.readthedocs.io/en/latest/guide/Guide_Review.html#reviewer-checklist) passed
- [ ] [Tests][testing] passing
- [ ] No new [linter issues][linter]

[contrib]: https://github.com/CLIMADA-project/climada_python/blob/develop/CONTRIBUTING.md
[testing]: https://climada-python.readthedocs.io/en/latest/guide/Guide_Testing.html
[linter]: https://climada-python.readthedocs.io/en/latest/guide/Guide_continuous_integration_GitHub_actions.html#static-code-analysis

### CODE REVIEW

**Comment by spjuhel** [comment]
Should we also mention this in the programming guidelines?

**Comment by chahank** [comment]
Yes!

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,11 +13,12 @@ Code freeze date: YYYY-MM-DD
...
 - `climada.util.interpolation` module for inter- and extrapolation util functions used in local exceedance intensity and return period functions [#930](https://github.com/CLIMADA-project/climada_python/pull/930)
-  
+
 ### Changed
 
 - In `climada.util.plot.geo_im_from_array`, NaNs are plotted in gray while cells with no centroid are not plotted [#929](https://github.com/CLIMADA-project/climada_python/pull/929)
 - Renamed `climada.util.plot.subplots_from_gdf` to `climada.util.plot.plot_from_gdf` [#929](https://github.com/CLIMADA-project/climada_python/pull/929)
+- Opening xarray dataset files is now done with a context manager (`with xr.open_dataset(path) as data:`) [#953](https://github.com/CLIMADA-project/climada_python/pull/953)
 
 ### Fixed
```
Comment (emanuel-schmid) [inline CHANGELOG.md:24]:
```suggestion

-  Closed some file handles left open when reading netcdf files with `climada.hazard` modules [#953](https://github.com/CLIMADA-project/climada_python/pull/953)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -13,11 +13,12 @@ Code freeze date: YYYY-MM-DD
 ### Added
 
 - `climada.util.interpolation` module for inter- and extrapolation util functions used in local exceedance intensity and return period functions [#930](https://github.com/CLIMADA-project/climada_python/pull/930)
-  
+
 ### Changed
 
 - In `climada.util.plot.geo_im_from_array`, NaNs are plotted in gray while cells with no centroid are not plotted [#929](https://github.com/CLIMADA-project/climada_python/pull/929)
 - Renamed `climada.util.plot.subplots_from_gdf` to `climada.util.plot.plot_from_gdf` [#929](https://github.com/CLIMADA-project/climada_python/pull/929)
+- Opening xarray dataset files is now done with a context manager (`with xr.open_dataset(path) as data:`) [#953](https://github.com/CLIMADA-project/climada_python/pull/953)
```
Comment (emanuel-schmid) [inline CHANGELOG.md:21]:
```suggestion
```

**Review by emanuel-schmid — COMMENTED** [review]
Very valuable contribution. 🙌 

I'd just change the changelog 

1) this is a fix not a change
2) the log should tell what has been done, but not how. If somebody is interested they can click on the PR link.

**Pull request merged**